src/sql/src/plan.rs

```diff
@@ -709,7 +709,7 @@ pub struct SetTransactionPlan {
 
 #[derive(Clone, Debug)]
 pub struct SelectPlan {
-    pub source: MirRelationExpr,
```

Comment:
I think in general we want planning to do the AST->HIR->MIR work. Moving HIR->MIR out of planning and into sequencing has an unknown risk, and the adapter team currently doesn't have enough knowledge about how planning works to give an informed opinion here.

Reply:
If we wanted to keep the MIRs here, what would have to change instead to make this PR possible?

Reply:
@mjibson Let's move the discussion in #24581. I tried to put some motivating thoughts about this change in the PR description and will add you and @benesch as reviewers as soon as all PR and nightlies tests pass for this PR.